akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala

```diff
@@ -785,8 +785,14 @@ object PartitionHub {
    */
   @ApiMayChange
   def sink[T](partitioner: (Int, T) ⇒ Int, startAfterNrOfConsumers: Int,
-              bufferSize: Int = defaultBufferSize): Sink[T, Source[T, NotUsed]] =
-    statefulSink(() ⇒ (info, elem) ⇒ info.consumerIdByIdx(partitioner(info.size, elem)), startAfterNrOfConsumers, bufferSize)
+              bufferSize: Int = defaultBufferSize): Sink[T, Source[T, NotUsed]] = {
+    val fun: (ConsumerInfo, T) ⇒ Long = { (info, elem) ⇒
+      val idx = partitioner(info.size, elem)
+      if (idx < 0) -1L
```

Comment:
Why is this remapping of negative number to -1 needed?

Reply:
it's not needed, we could keep the negative idx as the id, but since idx and id are not treated as the same thing for the positive case we might as well map it to a single value